FueledUtils/UIExtensions.swift

```diff
@@ -164,11 +207,102 @@ public extension UIView {
+	/// This methods internally updates the following properties of the backing `CALayer` (`self.layer`):
+	/// - `CALayer.shadowColor`
+	/// - `CALayer.shadowOpacity`
+	/// - `CALayer.shadowOffset`
+	/// - `CALayer.shadowRadius`
+	/// - `CALayer.shadowPath`
+	///
+	/// When using this method, it is recommended **not** to modify any of these properties to avoid unexpected results.
+	///
+	/// - Parameters:
+	///   - color: The color of the shadow, as specified in Sketch (usually without the alpha component, specified afterwards)
+	///   - alpha: The alpha (opacity) of the shadow, as specified in Sketch
+	///   - xAxis: The X direction of the shadow, as specified in Sketch
+	///   - yAxis: The y direction of the shadow, as specified in Sketch
+	///   - blur: The blur offset of the shadow, as specified in Sketch
+	///   - spread: The spread of the shadow, as specified in Sketch.
+	///   - path: The path the shadow should use. If `nil`, it defaults to a rectangle of the size of the bounds of the receiver
+	///
+	/// - Note: It is safe to call this method multiple times without calling `removeSketchShadow` between each calls.
+	///
+	func applySketchShadow(
+		color: UIColor = .black,
+		alpha: Float = 0.5,
+		xAxis: CGFloat = 0.0,
+		yAxis: CGFloat = 2.0,
+		blur: CGFloat = 4.0,
+		spread: CGFloat = 0.0,
+		path: CGPath? = nil)
+	{
+		self.layer.shadowColor = color.cgColor
+		self.layer.shadowOpacity = alpha
+		self.layer.shadowOffset = CGSize(width: xAxis, height: yAxis)
+		self.layer.shadowRadius = blur / 2.0
+
+		if path == nil || spread == 0.0 {
+			self.layer.shadowPath = nil
+		} else {
+			let scaleFactor = (self.bounds.size.width + spread * 2.0) / self.bounds.size.width
+			let path = (path.map { UIBezierPath(cgPath: $0) } ?? UIBezierPath(rect: self.bounds))
+			path.apply(CGAffineTransform(scaleX: scaleFactor, y: scaleFactor))
+			self.layer.shadowPath = path.cgPath
+		}
+	}
+
+	///
+	/// Remove a shadow as set by `applySketchShadow`
+	/// This method will reset any shadows sets on the backing layer, _even it wasn't applied by `applySketchShadow`
+	///
+	func removeSketchShadow() {
+		self.layer.shadowColor = nil
+		self.layer.shadowOpacity = 0.0
+		self.layer.shadowOffset = .zero
+		self.layer.shadowRadius = 0.0
+		self.layer.shadowPath = nil
+	}
+}
+
+@available(iOS 9.0, *)
+extension UIStackView {
+	///
+	/// Removes all of the current arranged subviews from the `UIStackView`, optionally
+	/// allowing to remove them from the `UIStackView`'s subviews as well.
+	///
+	/// - Parameters:
+	///   - removeFromHierachy: If `true`, each views is also removed from the receiver using `removeFromSuperview()`.
+	///     If `false`, `removeFromSuperview()` is not called.
+	///
+	func removeArrangedSubviews(removeFromHierachy: Bool) {
+		let arrangedSubviews = self.arrangedSubviews
+		arrangedSubviews.forEach { self.removeArrangedSubview($0, removeFromHierachy: true) }
```

Comment:
@stephanecopin I suppose this: `arrangedSubviews.forEach { $0.removeFromSuperview() }` will remove the views as subviews and the UIStackView will remove them from its arrangedSubviews list automatically.

Reply:
@heymansmile it does indeed, I've updated the method in consequence. I think keeping the methods does help, as it allows to explicitely keep the semantics between "arranged subviews" and "subview" different. It's definitely happen to me to add just subviews to a stack view (to add a background color for example).

Reply:
@stephanecopin Ok, but I'm just curious when do you really need such call:
`.removeArrangedSubviews(removeFromHierachy: false)` ?
I mean if you need to add a background color it's unlikely that you at first add a view to arranged views and then remove this from arranged and leave as a subview?

Reply:
@heymansmile You would never use it, it's just to make it clear that if you use that it won't remove it from the superview list (you can't overload methods with the same name; and I don't think swizzling is a good idea here as it would change the default behavior).
The goal here is to make it explicit that the subview can be removed from the hierarchy, and through autocompletion prompts the user to ask themselves "what's the default behavior", with is what I'm aiming for.

Reply:
@stephanecopin btw was it intended to ignore the argument and explicitly pass `true` here
`arrangedSubviews.forEach { self.removeArrangedSubview($0, removeFromHierachy: true) }` ?

Reply:
@heymansmile it's intended to use the argument and pass `true` when using it

Reply:
@stephanecopin I'm confused: the argument is not used at all, is it? So maybe change the signature to something like that `func removeArrangedSubviewsIncludingHierarchy()` ?

Reply:
It is used, if `false` then it's equivalent to calling it with no arguments

Reply:
I decided against naming the method that way, just for consistency with the original name (and having greater flexibility)

Reply:
@heymansmile I guess my confusion added to yours, I fixed the method so it uses the argument passed in the method as it was originally meant to. Resolving this conversation 😉

Reply:
;)